internal/upf/upf.go

```diff
@@ -118,6 +136,31 @@ func (u *UPF) Close() {
 	logger.UpfLog.Info("UPF resources released")
 }
 
+func (u *UPF) Reload(masquerade bool) error {
+	u.bpfObjects.Masquerade = masquerade
+
+	err := u.bpfObjects.Load()
```

Comment:
Does this overwrite the program if it is already loaded?

Reply:
Yes, going from NAT to non NAT and vice versa loads the program correctly.